scriptworker/gpg.py

```diff
@@ -1113,25 +1131,337 @@ def rebuild_gpg_home_signed(context, real_gpg_home, my_pub_key_path,
+        gpg: next trustdb check due at 2018-09-17
+        gpg: Good signature from "Aki Sasaki (2016.09.16) <aki@escapewindow.com>" [ultimate]
+        gpg:                 aka "Aki Sasaki (2016.09.16) <aki@mozilla.com>" [ultimate]
+        gpg:                 aka "Aki Sasaki (2016.09.16) <asasaki@mozilla.com>" [ultimate]
+        gpg:                 aka "[jpeg image of size 5283]" [ultimate]
+        Author: Aki Sasaki <aki@escapewindow.com>
+        Date:   Mon Sep 19 21:50:35 2016 -0700
+
+            add another check + small fixes + comments
+
+    or::
+
+        commit 02dc29251021519ebac4508545477a7b23efea49
+        gpg: Signature made Tue Sep 20 04:22:57 2016 UTC
+        gpg:                using RSA key 0xFC829B7FFAA9AC38
+        gpg: Good signature from "Aki Sasaki (2016.09.16) <aki@escapewindow.com>"
+        gpg:                 aka "Aki Sasaki (2016.09.16) <aki@mozilla.com>"
+        gpg:                 aka "Aki Sasaki (2016.09.16) <asasaki@mozilla.com>"
+        gpg:                 aka "[jpeg image of size 5283]"
+        Author: Aki Sasaki <aki@escapewindow.com>
+        Date:   Mon Sep 19 21:22:40 2016 -0700
 
-    There are a number of ways to do this.  `git show --show-signature` will
-    show the output of `gpg --verify`, assuming we have the key already marked
-    valid in our web of trust, also assuming we're using the proper gpg_home.
+            add travis tests for commit signatures.
 
-    This function allows for unsigned commits between signed commits.
-    We may disallow those in the future.
 
     Args:
-        gpg (gnupg.GPG): the GPG instance.
-        output (str): the output of `git log --format='%H:%GK'`
-        trusted_fingerprints (list): the gpg fingerprints of valid keys.
+        output (str): the output from `git log --no-merges -n 1 --show-signature`
+
+    Raises:
+        ScriptWorkerGPGException: on error.
+    """
+    BAD = {
+        "gpg: Can't check signature:": "",
+        "gpg: WARNING: This key is not certified with a trusted signature!": "",
+    }
+    GOOD = re.compile(r"""^gpg: Good signature from ".*"( \[(ultimate|trusted)\])?$""")
+    messages = []
+    lines = output.splitlines()
+    status = False
+    for line in lines:
+        m = GOOD.match(line)
+        if m is not None:
+            status = True
+            continue
+        for k, v in BAD.items():
+            if line.startswith(k):
+                messages.append(v)
+                continue
+    if not status:
+        messages.append("No trusted signature!")
+    if messages:
+        raise ScriptWorkerGPGException("\n".join(messages + ["output:", output]))
+
+
+async def get_git_revision(path, exec_function=asyncio.create_subprocess_exec):
+    """Get the git revision of path.
+
+    Args:
+        path (str): the path to run `git rev-parse HEAD` in.
+
+    Returns:
+        str: the revision.
+
+    Raises:
+        ScriptWorkerRetryException: on failure.
+    """
+    proc = await exec_function(
```

Comment:
Hm.  Makes sense, but I don't think it's pressing just to replace a single call.